src/main.rs

```diff
@@ -36,95 +60,128 @@ fn main() {
         (404, "Not Found"),
         (405, "Method Not Allowed"),
         (500, "Internal Server Error"),
-    ]);
-
-    // define routes and their handlers
+    ]));
     let routes = Arc::new(HashMap::from([(
         "/",
         PathProperties {
-            handler: handle_path_status as fn(AppState) -> (),
+            handler: handle_path_status as fn(RequestState) -> (),
             method_conent_map: HashMap::from([("GET", "application/json")]),
         },
     )]));
+    let request_cache = Arc::new(Mutex::new(HashMap::new()));
 
-    // setup server
+    // create server
     let server = TcpListener::bind("127.0.0.1:3000").expect("Could not bind to address");
 
-    // handle incoming connections
-    let thread_pool = thread::spawn(move || {
-        for stream in server.incoming() {
+    // handle connections
+    let request_cache_clone = Arc::clone(&request_cache);
+    let connection_pool = thread::spawn(move || {
+        for stream_result in server.incoming() {
```

Comment:
> this code is ass
 
"i know your honor"

> then why merge it

"it's the best i can do your honor"

> YOU'RE FIRED 

NO YOUR HONOR I HAVE FAMILY TO FEED